iscservice/demo/util.h

```diff
@@ -0,0 +1,158 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+
+#ifndef __MOTR_ISCSERVICE_DEMO_UTIL_H__
+#define __MOTR_ISCSERVICE_DEMO_UTIL_H__
+
+#include "iscservice/isc.h"  /* m0_fop_isc */
+#include "rpc/rpc_machine.h" /* M0_RPC_DEF_MAX_RPC_MSG_SIZE */
+
+/* Import */
+struct m0_fid;
+struct isc_req;
+struct m0_buf;
+struct m0_rpc_link;
+struct m0_layout_io_plop;
+struct m0_obj;
+struct m0_config;
+struct m0_client;
+enum   m0_conf_service_type;
+
+enum isc_buffer_len {
+	/** 32K */
+	CBL_DEFAULT_MAX = M0_RPC_DEF_MAX_RPC_MSG_SIZE >> 2,
+	CBL_DEFAULT_MIN = 256,
+};
+
+/** A request holding all parameters relevant to a computation. */
+struct isc_req {
+	/** Buffer to store returned result. */
+	struct m0_buf          cir_result;
+	/** Error code for the computation. */
+	int                    cir_rc;
+	/** RPC session of the ISC service. */
+	struct m0_rpc_session *cir_rpc_sess;
+	/** FOP for ISC service. */
+	struct m0_fop_isc      cir_isc_fop;
+	/** A generic fop for ISC service. */
+	struct m0_fop          cir_fop;
+	/** plop this request corresponds to. */
+	struct m0_layout_plop *cir_plop;
+	/** all reqs list link */
+	struct m0_list_link    cir_link;
+};
+
+extern struct m0_realm     uber_realm;
+extern struct m0_semaphore isc_sem;
+extern struct m0_list      isc_reqs;
+
+/**
+ * Initialise Motr-related stuff.
+ */
+int  isc_init(struct m0_config*, struct m0_client**);
+void isc_fini(struct m0_client*);
+
+/**
+ * Return parity group size for object.
+ */
+uint64_t isc_m0gs(struct m0_obj*, struct m0_client*);
+
+#define isc_reqs_teardown(req) \
```

Comment:
A generic version of m0_list_teardown(head, obj) can be added to lib/list.h

Reply:
Ok, added.